rtos/rtx/TARGET_CORTEX_M/RTX_CM_lib.h

```diff
@@ -545,28 +551,49 @@ extern uint32_t          __end__[];
 #define HEAP_START      (__end__)
 #elif defined(__ICCARM__)
 #pragma section="HEAP"
-#define HEAP_END  (void *)__section_end("HEAP")
+#pragma section="CSTACK"
 #endif
 
-void set_main_stack(void) {
-    uint32_t interrupt_stack_size = ((uint32_t)OS_MAINSTKSIZE * 4);
+extern unsigned char *mbed_heap_start;
+extern uint32_t mbed_heap_size;
+
+unsigned char *mbed_stack_isr_start = 0;
+uint32_t mbed_stack_isr_size = 0;
+
+/*
+ * set_stack_heap purpose is to set the following variables:
+ * -mbed_heap_start
+ * -mbed_heap_size
+ * -mbed_stack_isr_start
+ * -mbed_stack_isr_size
+ */
+void set_stack_heap(void) {
+
 #if defined(__ICCARM__)
-	/* For IAR heap is defined  .icf file */
-	uint32_t main_stack_size = ((uint32_t)INITIAL_SP - (uint32_t)HEAP_END) - interrupt_stack_size;
+
+    // Fixed sized heap
```

Comment:
Use ``/* */`` for consistency.

Shall we provide here some more detailed description why it is for IAR different than for other toolchains? (explains why there's #ifdef IAR), the comments are fine that its fixed but a reader might not know why its fixed?

```
/* IAR does not support floating heap and stack, stack and heap are fixed sized - defined in the linker command file.
 */
```